bokeh/models/sources.py

```diff
@@ -38,6 +39,10 @@ class DataSource(PlotObject):
                 hit/selected
     """)
 
+    callback = Instance(Callback, help="""
+    A callback to run in the browser whenever the selection is changed.
+    """)
+
```

Comment:
It's a good thought.  Is the intent in this PR/diff to just handle the selection change?  There are really two options: having two different callbacks, or having a single callback with multiple kinds of events.  I tend to prefer the latter option, because it then makes it easier to write things that intercept or chain or change the events.